br/pkg/lightning/backend/local/local.go

```diff
@@ -267,11 +267,30 @@ func (g *targetInfoGetter) CheckRequirements(ctx context.Context, checkCtx *back
 	if err := tikv.CheckTiKVVersion(ctx, g.tls, g.pdAddr, localMinTiKVVersion, localMaxTiKVVersion); err != nil {
 		return err
 	}
+	storeAddr, err := FetchStoreAddr(ctx, db)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if err := tikv.CheckTiDBDestination(ctx, g.tls, g.pdAddr, storeAddr); err != nil {
+		return err
+	}
 
 	serverInfo := version.ParseServerInfo(versionStr)
 	return checkTiFlashVersion(ctx, g.targetDBGlue, checkCtx, *serverInfo.ServerVersion)
 }
 
+// FetchStoreAddr gets the stores address info from tidb server using query
+func FetchStoreAddr(ctx context.Context, db utils.QueryExecutor) (string, error) {
+	var StoreAddr string
+	const queryStoreAddr = "select distinct INSTANCE from INFORMATION_SCHEMA.CLUSTER_INFO where TYPE='tikv' or TYPE='tiflash' limit 1;"
```

Comment:
why `limit 1`

Reply:
i think `limit 1` is enough to confirm destination  correct. and,all address maybe too much.
if i collect all store addr from both pd api and cluster_info table,i should sort  and compare them.
a bit redundant

Reply:
but the `storeAddr` and the row picked out of `limit 1` is likely different

you'd better use `select 1 from information_schema.cluster_info where instance = ? and type in ('tikv', 'tiflash')` if you worry about the size of result set

Reply:
```
FetchStoreAddr --> RandStoreAddr
storeAddr --> randStoreAddr
```
What do you think?

Reply:
@kennytm  i've changed the logic using `select 1 from information_schema.cluster_info where instance = ? and type in ('tikv', 'tiflash')`. Please review it.